MediaAdmin/EventSubscriber/UpdateFolderPathSubscriber.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace OpenOrchestra\MediaAdmin\EventSubscriber;
+
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use OpenOrchestra\Media\Repository\FolderRepositoryInterface;
+use OpenOrchestra\MediaAdmin\Event\FolderEvent;
+use OpenOrchestra\MediaAdmin\FolderEvents;
+use OpenOrchestra\BaseBundle\Context\CurrentSiteIdInterface;
+
+/**
+ * Class UpdateFolderPathSubscriber
+ */
+class UpdateFolderPathSubscriber implements EventSubscriberInterface
+{
+    protected $folderRepository;
+    protected $eventDispatcher;
+    protected $currentSiteManager;
+
+    /**
+     * @param FolderRepositoryInterface  $nodeRepository
+     * @param EventDispatcherInterface $eventDispatcher
+     * @param CurrentSiteIdInterface   $currentSiteManager
+     */
+    public function __construct(
+        FolderRepositoryInterface $folderRepository,
+        EventDispatcherInterface $eventDispatcher,
+        CurrentSiteIdInterface $currentSiteManager
+    ){
+        $this->folderRepository = $folderRepository;
+        $this->eventDispatcher = $eventDispatcher;
+        $this->currentSiteManager = $currentSiteManager;
+    }
+
+    /**
+     * @param FolderEvent $event
+     */
+    public function updatePath(FolderEvent $event)
+    {
+        $folder = $event->getFolder();
+        $parentPath = $folder->getPath();
+        $siteId = $this->currentSiteManager->getCurrentSiteId();
+        $sons = $this->folderRepository->findByParentAndSite($folder->getId(), $siteId);
+
+        $sonsToUpdate = array();
+        foreach ($sons as $son) {
+            $son->setPath($parentPath . '/' . $son->getFolderId());
+            $sonsToUpdate[$son->getFolderId()] = $son;
```

Comment:
The 'one pass' update is better on a performance view you're right. But as several listeners could listen to FolderEvents::PATH_UPDATED, I'm not sure of the result if the tree is bypassed by a 'one pass'. It may have some colisions between listeners. By doing a recursion, each folder is update one after all. This should avoid listener collisions.

I agree qith the lack of unicity, I add a new task on the board for this